bump to libCEC 4.0.0 + some rebased PRs

opdenkamp opened the pull request:
Changes for the libCEC 4.0.0 bump and rebased PRs #8116 and #8549, so we don't have to wait for these to be fixed up again.

It's a major bump because of some interface changes. Mainly cleaned up old stuff that's no longer being used.

libCEC hasn't been tagged yet, just putting up the PR now for review and testing. libCEC's 4.0.0 "beta" is in the master branch

edit: fixed wrong pr linked (missing number)

popcornmix commented on system/peripherals.xml:
```diff
@@ -16,7 +16,7 @@
     <setting key="cec_standby_screensaver" type="bool" value="0" label="36009" order="6" />
     <setting key="cec_wake_screensaver" type="bool" value="1" label="36010" order="7" />
-    <setting key="standby_pc_on_tv_standby" type="enum" value="13011" label="36029" order="8" lvalues="36028|13005|13011" />
+    <setting key="standby_pc_on_tv_standby" type="enum" value="13011" label="36029" order="8" lvalues="36028|13005|13011|13009|36043|36045" />
```
36043 is "Number of buffers used by graphics driver" which doesn't sound right here

opdenkamp replied:
fixed. the code had the correct id, the xml didn't. thanks

wsnipex commented:
please bump version in https://github.com/xbmc/xbmc/blob/master/project/cmake/modules/FindCEC.cmake#L18

Glenn-1990 commented on xbmc/peripherals/devices/PeripheralCecAdapter.cpp:
```diff
@@ -614,43 +623,67 @@ void CPeripheralCecAdapter::SetMenuLanguage(const char *strLanguage)
+    g_application.ExecuteXBMCAction("Quit");
+    break;
+  case LOCALISED_ID_PAUSE:
+    m_bStarted = false;
```
Pause does not quit kodi or libcec, so this should be handled as stop below?

opdenkamp replied:
yup, you're right, thanks

opdenkamp replied:
fixed now

Glenn-1990 commented on xbmc/peripherals/devices/PeripheralCecAdapter.cpp:
```diff
@@ -1385,6 +1385,7 @@ void CPeripheralCecAdapter::SetConfigurationFromSettings(void)
   m_bPowerOffScreensaver               = GetSettingBool("cec_standby_screensaver") ? 1 : 0;
   m_bPowerOnScreensaver                = GetSettingBool("cec_wake_screensaver") ? 1 : 0;
   m_bSendInactiveSource                = GetSettingBool("send_inactive_source") ? 1 : 0;
+  m_configuration.bAutoWakeAVR         = GetSettingBool("power_avr_on_as") ? 1 : 0;
```
There is no SetSetting for this in SetConfigurationFromLibCEC(), not sure if that's really needed?

opdenkamp replied:
It is. This is read from Kodi's config only. I got rid of all the unneeded stuff in `SetConfigurationFromLibCEC()`

opdenkamp commented:
@wsnipex I've uploaded the debs to https://launchpad.net/~pulse-eight/+archive/ubuntu/libcec and updated FindCEC.cmake

wsnipex commented:
thanks, added it to our PPA as well

opdenkamp commented:
jenkins build this please

opdenkamp commented:
looks like the internal libCEC build from depends isn't happy yet. will check later this weekend

opdenkamp commented:
depends are building fine here and the build log is gone. jenkins build this please

opdenkamp commented:
platform lib in depends needs a bump too, will fix it

opdenkamp commented:
jenkins build this please

opdenkamp commented:
no idea what this travis thing thinks has failed, but jenkins is happy with it, so I'll clean this up and it's ready for merging then afaict

hudokkow commented:
Don't mind Travis-CI. It fails on a test with clang
> The following tests FAILED:
	393 - TestThreadLocal.Stack (SEGFAULT)

opdenkamp committed:
[cec] bump to libCEC 4.0.0

opdenkamp committed:
[cec] Added advanced settings for action when TV goes standby

Added advances settings for action when TV goes standby : Pause
playback, Stop playback, Exit Kodi

opdenkamp committed:
[cec] add CEC IMX adapter as known type and set no polling.

that eliminates extensive periodic /sys walkthrough

(this is reused Lars's commit e59d7e028288464e6890141a830e4a83d4b9d065)

opdenkamp committed:
[cec] added: setting to make libCEC wake up the AVR explicitly when activating the source.

ref: https://github.com/Pulse-Eight/libcec/issues/156

opdenkamp commented:
thanks @hudokkow 

PR rebased onto master, label id updated cause something else took it and squashed commits, so it's ready if jenkins is happy with it, so jenkins build this please

Rechi commented on system/peripherals.xml:
```diff
@@ -16,15 +16,15 @@
     <setting key="cec_hdmi_port" type="int" value="1" min="1" max="15" label="36015" order="13" />
     <setting key="physical_address" type="string" label="36021" value="0" order="14" />
-    <setting key="port" type="string" value="" label="36022" order="15" />
+    <setting key="power_avr_on_as" type="bool" label="36045" value="0" order="15" />
```
shouldn't this be `#36046`?

opdenkamp replied:
meeh damn magic numbers... you're right, will fix it tonight

ksooo replied:
Ping @opdenkamp. You wanted to fix this. ;-)

Rechi commented on system/peripherals.xml:
```diff
@@ -16,15 +16,15 @@
     <setting key="cec_standby_screensaver" type="bool" value="0" label="36009" order="6" />
     <setting key="cec_wake_screensaver" type="bool" value="1" label="36010" order="7" />
-    <setting key="standby_pc_on_tv_standby" type="enum" value="13011" label="36029" order="8" lvalues="36028|13005|13011" />
+    <setting key="standby_pc_on_tv_standby" type="enum" value="13011" label="36029" order="8" lvalues="36028|13005|13011|13009|36044|36046" />
```
shouldn't this be `#36045` instead of `#36046`?
